tests/testsuite/update.rs

```diff
@@ -1659,3 +1659,56 @@ fn update_with_missing_feature() {
         )
         .run();
 }
+
+#[cargo_test]
+fn update_incompatible() {
+    Package::new("compatible", "1.0.0").publish();
+    Package::new("incompatible", "1.0.0").publish();
+    Package::new("pinned", "1.0.0").publish();
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "bar"
+                version = "0.0.1"
+                edition = "2015"
+                authors = []
+
+                [dependencies]
+                compatible = "1.0"
+                incompatible = "1.0"
+                pinned = "<99.0"
+            "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    p.cargo("check").run();
+
+    Package::new("compatible", "1.0.1").publish();
+    Package::new("incompatible", "1.0.1").publish();
+    Package::new("pinned", "1.0.1").publish();
+
+    Package::new("compatible", "2.0.0").publish();
+    Package::new("incompatible", "2.0.0").publish();
+    Package::new("pinned", "2.0.0").publish();
+
+    p.cargo("update --incompatible")
+        .with_status(1)
+        .with_stderr_contains("[ERROR] unexpected argument '--incompatible' found")
+        .run();
+
+    p.cargo("update")
```

Comment:
We need to verify the manifests were updated

Reply:
The manifests are _correct_ now, but they don't preserve formatting as well as they should. Any ideas how to address this? I guess I should look more into cargo-edit.